curriculum/challenges/english/25-front-end-development/review-html-accessibility/671a87a39b88245a579c2271.md

```diff
@@ -9,7 +9,46 @@ dashedName: review-html-accessibility
+
+## Accessibility Auditing Tools
+
+- **Google Lighthouse**: an open-source tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+- **Wave**: a web accessibility evaluation tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+- **IBM Equal Accessibility Checker**: a web accessibility evaluation tool that can be used to audit the accessibility of a website. It provides a report with recommendations for improving accessibility.
+
+## Accessibility Best Practices
+
+- **Proper heading level structure**: You should use proper heading levels to create a logical structure for your content. This helps screen readers and other assistive technologies understand the content of your website.
+- **Accessibility and Tables**: When using tables, you should use the `<th>` element to define header cells and the `<td>` element to define data cells. This helps screen readers and other assistive technologies understand the structure of the table.
+- **Importance for inputs to have an associated label**: You should use the `<label>` element to associate labels with form inputs. This helps screen readers and other assistive technologies understand the purpose of the input.
+- **Importance of good `alt` text**: You should use the `alt` attribute to provide a text alternative for images. This helps screen readers and other assistive technologies understand the content of the image.
+- **Importance of good link text**: You should use descriptive link text to help users understand the purpose of the link. This helps screen readers and other assistive technologies understand the content of the link.
+- **Best practices for making audio and video content accessible**: You should provide captions and transcripts for audio and video content to make it accessible to people with hearing impairments. You should also provide audio descriptions for video content to make it accessible to people with visual impairments.
+- **`tabindex` attribute**: used to specify the order in which elements should receive focus when using the tab key to navigate the website. This can help users with mobility impairments navigate the website more easily.
+- **`accesskey` attribute**: used to define a keyboard shortcut for an element. This can help users with mobility impairments navigate the website more easily.
+
+## WAI-ARIA, Roles, and Attributes
+
+- **WAI-ARIA**: a set of attributes that can be added to HTML elements to improve accessibility. It provides additional information to assistive technologies about the purpose and structure of the content.
+- **Aria roles**: a set of predefined roles that can be added to HTML elements to define their purpose. This helps screen readers and other assistive technologies understand the content of the website. Examples include `role="button"`, `role="dialog"`, and `role="navigation"`.
```

Comment:
Because I can't help it :smile: In a perfect world, there would never be the need to use `role="button"` because you would always use the `button` element instead. CSS is good enough nowadays that the arguments over being difficult to style a button are gone. The role of `dialog` is also going the same way now that the `dialog` element has full support in all browsers and is accessible out of the box. And of course the `nav` element has been around for a long time.

 If you want to humor me, you could use `role="tab", role="menu", and role="alert"` instead. These are all roles that don't have HTML equivalents. But I'm not insisting on it.

Reply:
I am looking at the associated lecture script and it looks like quite a few roles were mentioned. So we can update this review page to the three you mentioned.